### ISSUE TITLE
Installer exiting with status '14', even though RDP works with Remmina and Freerdp

### ISSUE THREAD

**Issue by AlbusPercivalDumbledore** [winapps-org/winapps#244]
> Attempting to establish a Remote Desktop connection with Windows... Failed!
> 
> ERROR: REMOTE DESKTOP PROTOCOL FAILURE.
> FreeRDP failed to establish a connection with Windows.

Host: EndeavourOS
guest: Windows 10 LTSC

I am able to use RDP using Remmina.

Log: Before I logged in and out using Remmina

> [22:48:15:167] [760038:000b98e9] [ERROR][com.freerdp.core.nego] - [nego_process_negotiation_failure]: Error: SSL_CERT_NOT_ON_SERVER
> 
> [22:48:15:168] [760038:000b98e9] [ERROR][com.freerdp.core.nego] - [nego_process_negotiation_failure]: Error: SSL_CERT_NOT_ON_SERVER
> 
> [22:48:15:181] [760038:000b98e9] [WARN][com.freerdp.core.connection] - [rdp_client_connect_auto_detect]: messageChannelId == 0
> 
> [22:48:15:181] [760038:000b98e9] [WARN][com.freerdp.core.license] - [license_read_binary_blob_data]: license binary blob::type BB_ERROR_BLOB, length=0, skipping.
> 
> [22:48:15:181] [760038:000b98e9] [WARN][com.freerdp.core.connection] - [rdp_client_connect_auto_detect]: messageChannelId == 0
> 
> [22:48:15:195] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [Stream_CheckAndLogRequiredLengthWLogExVa]: [rdp_recv_font_map_pdu(/usr/src/debug/freerdp/FreeRDP-3.8.0/libfreerdp/core/activation.c:533)] invalid length, got 0, require at least 8 [element size=1]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 0: dli_fname=/usr/lib/libwinpr3.so.3 [0x7bbd768ea000], dli_sname=winpr_backtrace [0x7bbd76931b50]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 1: dli_fname=/usr/lib/libwinpr3.so.3 [0x7bbd768ea000], dli_sname=winpr_log_backtrace_ex [0x7bbd769320a0]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 2: dli_fname=/usr/lib/libwinpr3.so.3 [0x7bbd768ea000], dli_sname=Stream_CheckAndLogRequiredLengthWLogExVa [0x7bbd76932e20]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 3: dli_fname=/usr/lib/libwinpr3.so.3 [0x7bbd768ea000], dli_sname=Stream_CheckAndLogRequiredLengthEx [0x7bbd76933060]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 4: dli_fname=/usr/lib/libfreerdp3.so.3 [0x7bbd76a00000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 5: dli_fname=/usr/lib/libfreerdp3.so.3 [0x7bbd76a00000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 6: dli_fname=/usr/lib/libfreerdp3.so.3 [0x7bbd76a00000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 7: dli_fname=/usr/lib/libfreerdp3.so.3 [0x7bbd76a00000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 8: dli_fname=/usr/lib/libfreerdp3.so.3 [0x7bbd76a00000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 9: dli_fname=/usr/lib/libfreerdp3.so.3 [0x7bbd76a00000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 10: dli_fname=/usr/lib/libfreerdp3.so.3 [0x7bbd76a00000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 11: dli_fname=/usr/lib/libfreerdp3.so.3 [0x7bbd76a00000], dli_sname=freerdp_check_fds [0x7bbd76ab7130]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 12: dli_fname=/usr/lib/libfreerdp3.so.3 [0x7bbd76a00000], dli_sname=freerdp_check_event_handles [0x7bbd76abf3e0]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 13: dli_fname=xfreerdp3 [0x6549c3bb1000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 14: dli_fname=/usr/lib/libwinpr3.so.3 [0x7bbd768ea000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 15: dli_fname=/usr/lib/libc.so.6 [0x7bbd764c9000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 16: dli_fname=/usr/lib/libc.so.6 [0x7bbd764c9000], dli_sname=(null) [(nil)]
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [winpr_log_backtrace_ex]: 17: unresolvable, address=(nil)
> 
> [22:48:15:196] [760038:000b98e9] [WARN][com.freerdp.core.activation] - [rdp_recv_font_map_pdu]: [MS-RDPBCGR] 2.2.1.22.1 Font Map PDU Data (TS_FONT_MAP_PDU) paylaod size is 0 instead of 8

LOG: After Logging in and out through Remmina

> 
> [23:04:19:391] [784940:000bfa30] [ERROR][com.freerdp.core.nego] - [nego_process_negotiation_failure]: Error: SSL_CERT_NOT_ON_SERVER
> 
> [23:04:19:392] [784940:000bfa30] [ERROR][com.freerdp.core.nego] - [nego_process_negotiation_failure]: Error: SSL_CERT_NOT_ON_SERVER
> 
> [23:04:19:402] [784940:000bfa30] [ERROR][com.freerdp.core.transport] - [transport_read_layer]: BIO_read returned a system error 11: Resource temporarily unavailable
> 
> [23:04:19:402] [784940:000bfa30] [ERROR][com.freerdp.core] - [transport_read_layer]: ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
> 
> [23:04:19:344] [784940:000bfa30] [INFO][com.freerdp.client.x11] - [xf_setup_x11]: Enabling X11 debug mode.

Thanks for any help.

**Comment by AlbusPercivalDumbledore** [comment]
Freerdp works with /cert:ignore flag. 

`xfreerdp3 /u:name /p:pass /v:ip /cert:ignore`

Is there anyway to use this flag while installing winapps?

**Comment by oskardotglobal** [comment]
You should be able to put this into the `RDP_FLAGS` field of the config

**Comment by AlbusPercivalDumbledore** [comment]
> You should be able to put this into the `RDP_FLAGS` field of the config

Thanks! Totally missed that.

Using that flag didn't make any difference. Log shows same error as above.

**Comment by itiligent** [comment]
I've had a bunch of issues like this. I found you can trip yourself up and get this error if you do things like install winapps before rebooting or by changing the windows system name after you;ve win the winapps installer at least once. Try deleting any pre-existing certs in the $HOME/.config/freerdp/server might also help.

**Comment by AlbusPercivalDumbledore** [comment]
> I've had a bunch of issues like this. I found you can trip yourself up and get this error if you do things like install winapps before rebooting or by changing the windows system name after you've already run the winapps installer at least once. Try deleting any pre-existing certs in the $HOME/.config/freerdp/server might also help.

'~/.config/freerdp/server' directory doesn't exist.

I was desperate to have this running, so I've tried with both kvm and virtualbox, installed a different version of windows, wiped all prior trace of winapps. Nothing worked.

I also scoured freerdp issue tracker; tried a bunch of random things suggested there. Was able to bring down the errors to these:

> [03:41:06:223] [1847567:001c3114] [ERROR][com.freerdp.crypto] - [freerdp_tls_handshake]: BIO_do_handshake failed
> [03:41:06:223] [1847567:001c3114] [ERROR][com.freerdp.core] - [transport_default_connect_tls]: ERRCONNECT_TLS_CONNECT_FAILED [0x00020008]
> [03:41:06:134] [1847567:001c3114] [INFO][com.freerdp.client.x11] - [xf_setup_x11]: Enabling X11 debug mode.

Haven't had any luck in getting rid of these.

**Comment by itiligent** [comment]
Errors point towards something fishy with your TLS libraries or local TLS settings or networoking becasue your system is not able to negotiate a TLS session.. Its hard to say more are you have not provided any more detail on the enviroment.  Because Remina works might be throwing you off a bit - remember Remina logs also show it is also failing to use TLS.  

Do you have any Docker network weird port forwarding?, ipv6 enabled? You need to eliminate suspects more as I dont think your issue is WinApps based on what can be seen so far.. Run up a blank windows VM and point freeRDP3 to it with: 
`xfreerdp3 /cert:ignore /d: /u:username /p:password /v:container_ip`

if that works, then its between your host and winapps virtual platform, if it does not work, then its maybe more realted to a TLS library thing.

**Comment by maurocolella** [comment]
Line 1044 of setup.sh has multiple problems. I got a little further using `libvirt` as the backend with:

```
    $FREERDP_COMMAND \
        /cert:tofu \
        /d:"$RDP_DOMAIN" \
        /u:"$RDP_USER" \
        /p:"$RDP_PASS" \
        /sec:tls \
        /sec:nla \
        /scale:"$RDP_SCALE" \
        +auto-reconnect \
        +home-drive \
        -wallpaper \
        +dynamic-resolution \
        /app:program:"C:\Windows\System32\cmd.exe",cmd:"/C type NUL > ""$TEST_PATH"" && tsdiscon" \
        /v:"$RDP_IP" &>"$FREERDP_LOG" &
 ```
 
 Notice:
 ```
         /sec:tls \
        /sec:nla \
 ```
 
 This seems required for a Win11 guest, which I tested using a simpler, manual freerdp command.
 
 Notice also `""$TEST_PATH"" ` instead of `$TEST_PATH_WIN`.
 
 TL;DR this section of the setup code is currently buggy. It should include any custom RDP flags, test for the correct path, etc.
